pimd/pim6_cmd.c

```diff
@@ -448,6 +448,186 @@ DEFPY (no_ipv6_pim_rp_prefix_list,
 	return pim_process_no_rp_plist_cmd(vty, rp_str, plist);
 }
 
+DEFPY (interface_ipv6_mld_join,
+       interface_ipv6_mld_join_cmd,
+       "ipv6 mld join X:X::X:X$group [X:X::X:X$source]",
+       IPV6_STR
+       IFACE_MLD_STR
+       "MLD join multicast group\n"
+       "Multicast group address\n"
+       "Source address\n")
+{
+	char xpath[XPATH_MAXLEN];
+	struct prefix_ipv6 p;
+
+	if (source_str) {
+		(void)str2prefix_ipv6(source_str, &p);
```

Comment:
`DEFPY` gives you a `struct prefix_ipv6 source` parameter here, the `str2prefix_ipv6` is redundant

You can see the list of parameters and their types for a function like this:
```
$ lib/clippy python/clidef.py --show pimd/pim_cmd.c >/dev/null
"show ip pim [vrf NAME] join [A.B.C.D$s_or_g [A.B.C.D$g]] [json$json]":
	const char *               opt  vrf
	. . . struct in_addr       opt  s_or_g
	. . . struct in_addr       opt  g
	const char *               opt  json

"show ip pim [vrf NAME] jp-agg":
	const char *               opt  vrf

"show ip pim [vrf NAME] upstream [A.B.C.D$s_or_g [A.B.C.D$g]] [json$json]":
	const char *               opt  vrf
	. . . struct in_addr       opt  s_or_g
	. . . struct in_addr       opt  g
	const char *               opt  json

[…]
```
(the `_str` parameter is not listed explicitly and is added for all parameters that are not `const char *`.)

Reply:
Fixed